lib/include/modelSpec.h

```diff
@@ -179,6 +194,34 @@ class NNmodel
         }
 
         // Add neuron group
+        int MPIHostID = 0;
+#ifdef MPI_ENABLE
+        MPI_Comm_rank(MPI_COMM_WORLD, &MPIHostID);
+#endif
+        bool isLocal = (hostID == MPIHostID) && (deviceID == 0);
```

Comment:
Theoretically, I also prefer 2 because it seems clearly using different mapping in the different situation. But yes it will introduce more structural changes in the code. Maybe I should try 1 first?